### PR TITLE
Azure WAN swagger

### PR DESCRIPTION
**Pull request by priyanag**
Reopening the closed PR https://github.com/Azure/azure-rest-api-specs/pull/1170

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by priyanag** [comment]
@sergey-shandar I have edited the objects to have all the fields defined within properties. Kindly review.

File: arm-network/2017-03-01/swagger/azureWAN.json
Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Are there list of supported algorithms ? If so, please add them in description for easy discovery to customers.

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Are there list of supported algorithms ? If so, please add them in description for easy discovery to customers.

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Are there list of supported algorithms ? If so, please add them in description for easy discovery to customers.

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Are there list of supported algorithms ? If so, please add them in description for easy discovery to customers.

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: unnecessary empty line

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: please provide accurate and complete description

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Are all the properties of this model `patchable` ? I see that you are using them in `path` operation's body parameter as well.

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: What are the required body parameters for the `PUT` operation here?

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: `scalableVpnGateway` seems like a valid resource then why is this service not using the model in the array here? 

Customers have complained a lot about the pain of traversing things by names, instead it'd greatly simplify experience bf at least we can give then `id` of the subResource then it's easy to get that resource when required via `Microsoft.Resource` service.

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: same question as above here

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: same question as above

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: why not derive this model from `Resource` ?

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: why does `PUT` on `/subscriptions/{subscriptionId}/resourceGroup/{resourceGroupName}/providers/Microsoft.Network/hubs/{hubName}` is asking `name` - Hub name parameter twice. Once in path and other in body?

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: What are the required and readOnly parameters?

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: why only name and not model of `vNetPeeringConnection` resource which seems a valid resource?

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: please remove empty line

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: please remove empty line

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: please remove empty line

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: please remove empty line - from everywhere

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: seems like readOnly as used in only `GET` operation

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: seems like readOnly as used in only `GET` operation

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: why not derive this from `Resource` model?

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: what are the `readOnly` parameters for `PUT` / `PATCH`?

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: what are the `required` parameters for `PUT` / `PATCH`?

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: what are `readOnly` and `required` parameters?

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: what are `readOnly` and `required` parameters for `PUT` / `PATCH` operations ?

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: why is the service asking this parameter twice for `put` & `patch` operation ? Is one in body supposed to be `readOnly` ?

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Please specify `readOnly` & `required` parameters accurately for this model.

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Please specify `readOnly` & `required` parameters accurately for this model.

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: asking this parameter twice?

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Please specify `readOnly` & `required` parameters accurately for this model.

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Please specify `readOnly` & `required` parameters accurately for this model.

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: asking this parameter twice?

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Please specify `readOnly` & `required` parameters accurately for this model.

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: can this be negative ? Are there a rage of valid values?

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: please please no empty lines please :)

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Please specify `readOnly` & `required` parameters accurately for this model.

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Why ask this parameter twice?

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Please specify `readOnly` & `required` parameters accurately for this model.

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: I believe that i saw these properties in above models as well, feel free to create separate model if that conceptually works out for you.

Comment (vishrutshah) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Please specify `readOnly` & `required` parameters accurately for this model.

**Review by vishrutshah — CHANGES_REQUESTED** [review]

**Comment by vishrutshah** [comment]
@priyanag Here are some of the general comments to be addressed:

- [ ] Please provide x-ms-examples for the new service endpoints. That is the requirement. https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/x-ms-examples.md#why-x-ms-examples. **Tips**: If you have .net tests for the service, you can leverage the tool that is in the still development at https://github.com/Azure/oav/pull/48/files

- [ ] Please mark all long-running operation correctly using extension [x-ms-long-running-operation](https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-long-running-operation)

- [ ] Please add `Default` response code for all the operation's response sections. That is the way autorest handles the exceptions. Rule: ` M2049` : https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md

- [ ] Please let us know is there any specific reason you are modeling all the operation with atleast one of the negative response codes that look inconstant with the other swagger that exists in this `Microsoft.Network`. Reference: https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/creating-swagger.md#NegativeResponses

- [ ] Why do you model all the `list` operation with response code of `204` instead of returning `empty list` for `200` ? That also seems inconsistent with network clients.

- [ ] Is there any specific reason for adding prefix `Azure` in each operation name's NOUN part of NOUN_VERB pattern? Please remove it

- [ ] please update all the  models to accurately represent `readOnly` & `required` parameters. 

**Tips:** Leverage autorest linter to validate & fix other errors. Please see the build https://travis-ci.org/Azure/azure-rest-api-specs/jobs/232993239#L1082

Thanks

**Comment by vishrutshah** [comment]
@priyanag Let us know if you have an updates or Let us know if you need any help. Thanks!

File: arm-network/2017-03-01/swagger/azureWAN.json
Comment (priyanag) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Addressed

File: arm-network/2017-03-01/swagger/azureWAN.json
Comment (priyanag) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Addressed

File: arm-network/2017-03-01/swagger/azureWAN.json
Comment (priyanag) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Addressed

File: arm-network/2017-03-01/swagger/azureWAN.json
Comment (priyanag) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Addressed

File: arm-network/2017-03-01/swagger/azureWAN.json
Comment (priyanag) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Addressed

File: arm-network/2017-03-01/swagger/azureWAN.json
Comment (priyanag) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Addressed

File: arm-network/2017-03-01/swagger/azureWAN.json
Comment (priyanag) [inline arm-network/2017-03-01/swagger/azureWAN.json]: In order to be consistent with all other Networking swaggers, removing 400 from delete responses. Believe, the default response will be returned as a result.

File: arm-network/2017-03-01/swagger/azureWAN.json
Comment (priyanag) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Addressed

File: arm-network/2017-03-01/swagger/azureWAN.json
Comment (priyanag) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Addressed

File: arm-network/2017-03-01/swagger/azureWAN.json
Comment (priyanag) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Addressed

File: arm-network/2017-03-01/swagger/azureWAN.json
Comment (priyanag) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Removing 400 as response for delete, as is the case with other Networking Apis

File: arm-network/2017-03-01/swagger/azureWAN.json
Comment (priyanag) [inline arm-network/2017-03-01/swagger/azureWAN.json]: @vishrutshah - All comments after this one show up at same line

Comment (priyanag) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Addressed

Comment (priyanag) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Addressed

Comment (priyanag) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Addressed

Comment (priyanag) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Addressed

Comment (priyanag) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Addressed

Comment (priyanag) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Addressed

Comment (priyanag) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Addressed

Comment (priyanag) [inline arm-network/2017-03-01/swagger/azureWAN.json]: Addressed

**Comment by vishrutshah** [comment]
@priyanag As we are having trouble with the rest of the comments, feel free to address as many as possible and please ping me/let me know once you have made that changes so only then I can start reviewing again to make sure I don't post comments while you are still addressing. Thanks!

**Comment by vishrutshah** [comment]
@priyanag Let us know when you have any updates on this. Thanks!

**Comment by priyanag** [comment]
@vishrutshah - We are ironing out some of the details internally. I'll be sending out an update within couple of days. Thanks!

**Comment by vishrutshah** [comment]
@priyanag Thanks for the heads-up.

**Comment by vishrutshah** [comment]
@priyanag Please resolve the conflicts in `compositeNetworkClient.json`. Alos ping me once ready for review.

**Review by vishrutshah — CHANGES_REQUESTED** [review]
@priyanag Thanks for the updates. i was reviewing the PR and seems like there are too many Errors. I was wondering whether did you know about the [VS Code Linter Extension](https://github.com/Azure/openapi-lint-extension) and did you try that out to find out what these error & warning are about? Let me know. 

If you haven't tried, I'd encourage you to try that out and let us know how it goes. If you need any help, let me know as well. 

Thanks!

File: arm-network/2017-06-01/examples/ExpressRouteConnectionDelete.json
Comment (vishrutshah) [inline arm-network/2017-06-01/examples/ExpressRouteConnectionDelete.json]: Should this be 2017-06-01?

**Comment by vishrutshah** [comment]
@priyanag Thanks a lot for working on these and reducing the errors & warnings drastically. As the change is big, it may take few days to complete the review. Feel free to ping me if you need anything urgently. Thanks!

**Comment by fearthecowboy** [comment]
ATTENTION REQUIRED.
-------------

We have changed the `default` branch in the `azure-rest-api-specs` repository to use the branch labeled `current`.

This PR is opened against the `master` branch.

You have two options: 
1. Keep this PR open, and complete the review process as normal, and once this PR is merged against the `master` branch, we (@olydis & @fearthecowboy)  will create a PR against the current branch and make any updates required to work in the newly reorganized default branch.  

2. Close this PR and create a new one. That would mean pulling in the changes from the new default branch `current` moving any files and switching to the configuration-based layout.  (You are more than welcome to choose option 2, but I don't expect anyone to do that)

Either way, the work you are currently doing against the previous default branch `master` will be the last PR against that branch that we will accept and all new PRs will go against the new default branch  `current`.

If you have any questions or concerns, ping @fearthecowboy or @olydis (or go ask on the `#swagger` channel in slack) ASAP and we can help you thru the process.

**Comment by vishrutshah** [comment]
@ravbhatnagar Let me know whether you got chance to look at this PR from ARM perspective and have gathered any feedback. Thanks!

File: arm-network/2017-06-01/swagger/azureWAN.json
Comment (ravbhatnagar) [inline arm-network/2017-06-01/swagger/azureWAN.json]: All RP specific properties barring etag needs to go inside the properties bag

Comment (ravbhatnagar) [inline arm-network/2017-06-01/swagger/azureWAN.json]: same comment as before about properties bag.

Comment (ravbhatnagar) [inline arm-network/2017-06-01/swagger/azureWAN.json]: Are these resources (scalableVpnGateways and scalableExpressRouteGateways) created in the PUT of /hubs/{hubName}?

Comment (ravbhatnagar) [inline arm-network/2017-06-01/swagger/azureWAN.json]: This is also a nested resource under hubs. Is this created during the create of hub?

Comment (ravbhatnagar) [inline arm-network/2017-06-01/swagger/azureWAN.json]: nest under properties bag

**Comment by ravbhatnagar** [comment]
@johanste / @NiklasGustafsson  - Please take a look at these APIs for AzureAPIReviewBoard sign off.

File: arm-network/2017-06-01/swagger/azureWAN.json
Comment (ravbhatnagar) [inline arm-network/2017-06-01/swagger/azureWAN.json]: Does delete on the hub delete the scalableExpressRouteGateways and scalableVpnGateways under it?

File: arm-network/2017-06-01/swagger/azureWAN.json
Comment (ravbhatnagar) [inline arm-network/2017-06-01/swagger/azureWAN.json]: camel casing -> hubVnetConnections

**Review by ravbhatnagar — CHANGES_REQUESTED** [review]

File: arm-network/2017-06-01/swagger/azureWAN.json
Comment (vishrutshah) [inline arm-network/2017-06-01/swagger/azureWAN.json]: @ravbhatnagar Looks like all the model with this linter error are using [Resource](https://github.com/Azure/azure-rest-api-specs/blob/master/arm-network/2017-06-01/swagger/network.json#L97) defined in `network.json` file but not marked as `readOnly:true`. Is there a reason not to do now?

**Review by vishrutshah — CHANGES_REQUESTED** [review]

File: arm-network/2017-06-01/swagger/azureWAN.json
Comment (vishrutshah) [inline arm-network/2017-06-01/swagger/azureWAN.json]: looks like the `Resource` model is not modeled as expected.

Comment (vishrutshah) [inline arm-network/2017-06-01/swagger/azureWAN.json]: looks like the `Resource` model is not modeled as expected.

File: arm-network/2017-06-01/swagger/azureWAN.json
Comment (ravbhatnagar) [inline arm-network/2017-06-01/swagger/azureWAN.json]: Yes, but since this model is used by NRP all over their APIs I didn't comment on making the change. We can work on getting this addressed later. Makes sense?

Comment (vishrutshah) [inline arm-network/2017-06-01/swagger/azureWAN.json]: Okay. Sounds good.

File: arm-network/2017-06-01/swagger/azureWAN.json
Comment (vishrutshah) [inline arm-network/2017-06-01/swagger/azureWAN.json]: line 1797: `BodyTopLevelProperties` `RPCViolation` `Error`

Top level properties should be one of name, type, id, location, properties, tags, plan, sku, etag, managedBy, identity. Model definition 'WAN' has extra properties ['provisioningState,disableVpnEncryption,hubs,vpnSites']

There are other 7 models with the same error. @ravbhatnagar As per the rule, we should move the extra properties into `properties -> properties` section. Right?

Comment (ravbhatnagar) [inline arm-network/2017-06-01/swagger/azureWAN.json]: yes, I have added the comments above to fix this.

**Comment by fearthecowboy** [comment]
Is this PR nearly ready to merge? I'd like to get the last of the PRs against `master` completed and merged so we're not in this state.

If this is going to take much longer, you may need to create a new PR against  the `current` branch.

**Comment by priyanag** [comment]
@fearthecowboy - I will try to complete this PR by EOD tomorrow. Thanks!

File: arm-network/2017-06-01/swagger/azureWAN.json
Comment (priyanag) [inline arm-network/2017-06-01/swagger/azureWAN.json]: Yes

File: arm-network/2017-06-01/swagger/azureWAN.json
Comment (priyanag) [inline arm-network/2017-06-01/swagger/azureWAN.json]: @ravbhatnagar Other networking swaggers use the Resource defined in network.json and in order to be aligned, I would prefer that we have the same inheritance here and then later fix the resource definition in network.json to mark these properties as read-only. Does that work?

File: arm-network/2017-06-01/swagger/azureWAN.json
Comment (priyanag) [inline arm-network/2017-06-01/swagger/azureWAN.json]: Same as above

Comment (priyanag) [inline arm-network/2017-06-01/swagger/azureWAN.json]: Same as above

File: arm-network/2017-06-01/swagger/azureWAN.json
Comment (priyanag) [inline arm-network/2017-06-01/swagger/azureWAN.json]: All the other arm-network swaggers do not have a default response for the APIs. None of them have error definitions or default responses. Should I also remove the default response, in order to be consistent?

File: arm-network/2017-06-01/swagger/azureWAN.json
Comment (priyanag) [inline arm-network/2017-06-01/swagger/azureWAN.json]: It was that earlier and then internally we had some discussion and concluded to rather make it a Boolean with the default behavior as disableVpnEncrytion = false, i.e., encrypted by default

File: arm-network/2017-06-01/swagger/azureWAN.json
Comment (priyanag) [inline arm-network/2017-06-01/swagger/azureWAN.json]: Correcting this

**Comment by vishrutshah** [comment]
@priyanag I see that you mentioned that CI is passing but somehow it's not detecting all the files. https://travis-ci.org/Azure/azure-rest-api-specs/jobs/256216998#L1248

**Comment by salameer** [comment]
ping @priyanag

File: arm-network/2017-06-01/swagger/azureWAN.json
Comment (ravbhatnagar) [inline arm-network/2017-06-01/swagger/azureWAN.json]: This should return an error object and the properties must comply with the error definition defined in ARM RPC.

File: arm-network/2017-06-01/swagger/azureWAN.json
Comment (ravbhatnagar) [inline arm-network/2017-06-01/swagger/azureWAN.json]: I know in NRP, the 202 async create pattern is widely used. But we recommend using the 201 + provisioning State + Azure-AsyncOperation header format for PUT/PATCH. Any specific reason why you use the 202 and cant use 201?

File: arm-network/2017-06-01/swagger/azureWAN.json
Comment (ravbhatnagar) [inline arm-network/2017-06-01/swagger/azureWAN.json]: Can virtualWANs be moved from one resource group to another or from one subscription to another? If yes, please add support for the Move API

File: arm-network/2017-06-01/swagger/azureWAN.json
Comment (ravbhatnagar) [inline arm-network/2017-06-01/swagger/azureWAN.json]: camelcase

Comment (ravbhatnagar) [inline arm-network/2017-06-01/swagger/azureWAN.json]: enum

Comment (ravbhatnagar) [inline arm-network/2017-06-01/swagger/azureWAN.json]: description missing

Comment (ravbhatnagar) [inline arm-network/2017-06-01/swagger/azureWAN.json]: name, etag, type and tags should be top level properties . Currently they are nested under "properties"

Comment (ravbhatnagar) [inline arm-network/2017-06-01/swagger/azureWAN.json]: please confirm whether these will be fully qualified ARM resource Ids? Whenever an ARM resource is referenced by another resource, it must be done using the ARM resource id. since you are doing it at many places in this spec, pleae confirm.

Comment (ravbhatnagar) [inline arm-network/2017-06-01/swagger/azureWAN.json]: You are returning these child resources when you do a GET on the parent (virtualHub). Wouldn't it suffice to just return the Ids of these child resources. If the user wants to GET all the scalableVPNGateways for a particular VirtualHub, you can provide an API for that. GET /virtualHubs/{hubName}/scalableVpnGateways which will return an array.

Comment (ravbhatnagar) [inline arm-network/2017-06-01/swagger/azureWAN.json]: You are returning these child resources when you do a GET on the parent (virtualHub). Wouldn't it suffice to just return the Ids of these child resources. If the user wants to GET all the scalableExpressRouteGateways for a particular VirtualHub, you can provide an API for that. GET /virtualHubs/{hubName}/scalableExpressRouteGateways which will return an array.

Comment (ravbhatnagar) [inline arm-network/2017-06-01/swagger/azureWAN.json]: enum?

**Review by ravbhatnagar — CHANGES_REQUESTED** [review]
@priyanag - I have added some comments. Since this is a big PR will lot of new resource types with dependencies on each other, I would recommend scheduling a meeting to review it. Although we are trying to move away from in person reviews as much as possible, I guess this is one case where an inperson review would be quicker and more effective. Please loop in Johan Stenberg, Niklas Gustafsson and myself for the review.

**Comment by vishrutshah** [comment]
@priyanag I've sent you the instructions on email and linter error report for you to analyze. Thanks!

**Comment by salameer** [comment]
@priyanag can you please address feedback and update on this PR to avid having it  closed due to lack of activity

**Comment by vishrutshah** [comment]
@salameer Just an FYI: 

ARM team, SDK Team and Service team is working closely on this PR. We had a sync up meeting today to approve this from the ARM perspective and follow ups on certain points. We'll be having another meeting to go over remaining discussion around Wednesday/Thursday. Based on that that this is not going to be closed soon (in a day / two). 

We'd still like to keep this PR open until all comments are addressed rather than opening new PR on `current` branch due to the nature of this new RP. Once we all agree & get ARM & SDK sign-off, then it'd be just a matter of time to create a new PR on `current` branch. That'd save reviewers and PR owners time as many comments are already made and getting addressed here :)

Let me know if you thing otherwise! 

Thanks!

**Review by vishrutshah — CHANGES_REQUESTED** [review]
@priyanag Did you receive any updates/recommendations from ARM regarding the `VpnSiteConfig` download config api ?

**Comment by veronicagg** [comment]
@priyang any updates?

**Comment by vishrutshah** [comment]
@priyang Let me know once you open PR against private repo so that we can close this one. Thanks!

**Comment by vishrutshah** [comment]
Closing this PR as it's moved to private repo. Thanks @priyang for moving it.

**Pull request closed without merging**
